Orange/widgets/visualize/utils/owlegend.py

```diff
@@ -643,28 +661,14 @@ def set_domain(self, domain):
         # The first and last values must represent the range, the rest should
         # be dummy variables, as they are not shown anywhere
         values = self.__range
-
-        start, end, pass_through_black = class_var.colors
-        # If pass through black, push black in between and add index to vals
-        if pass_through_black:
-            colors = [self._convert_to_color(c) for c
-                      in [start, '#000000', end]]
-            values.insert(1, -1)
-        else:
-            colors = [self._convert_to_color(c) for c in [start, end]]
-
-        self.set_items(list(zip(values, colors)))
+        self.set_items((values, class_var.palette))
 
     def set_items(self, values):
-        vals, colors = list(zip(*values))
-
-        # If the orientation is vertical, it makes more sense for the smaller
-        # value to be shown on the bottom
-        if self.orientation == Qt.Vertical and vals[0] < vals[len(vals) - 1]:
-            colors, vals = list(reversed(colors)), list(reversed(vals))
-
+        vals, palette = values
+        if self.orientation == Qt.Vertical:
+            vals = reversed(vals)
```

Comment:
I am always nervous with these one-time-use generator iterables.
While it does seem to be ok now, vals are passed on to `ContinuousLegendItem` and there to `_format_values` and no one guarantees that they will use vals only once (even though they do now).
How about keeping the `list( )` around `reversed`?